### ISSUE TITLE
sort_values() issue in FrameGO (not frame)

### ISSUE THREAD

**Issue by abefetterman** [InvestmentSystems/static-frame#86]
First off, this project is great, and much needed! Thanks!

I'm having an issue with calling sort_values on a `FrameGO`. Using static_frame v0.3.6. 

For example:
```
>>> f = sf.Frame(np.random.rand(5,2),columns=['a','b'])
>>> f = f.set_index('a')
>>> f.sort_values('b')
<Frame>
<Index>              a                    b                   <<U1>
<Index>
0.9914872606511099   0.9914872606511099   0.18321744935278328
0.06284940686768803  0.06284940686768803  0.23893175489779928
0.004295355913818244 0.004295355913818244 0.5492554550266754
0.35484130432136163  0.35484130432136163  0.7047498181734183
0.376705716066418    0.376705716066418    0.7816721126374693
<float64>            <float64>            <float64>
```

But for a FrameGO:
```
>>> f = sf.FrameGO(np.random.rand(5,2),columns=['a','b'])
>>> f = f.set_index('a')
>>> f.sort_values('b')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/static_frame/core/frame.py", line 2238, in sort_values
    blocks = self._blocks.iloc[order]
  File "/usr/local/lib/python3.6/site-packages/static_frame/core/util.py", line 1124, in __getitem__
    return self._func(key)
  File "/usr/local/lib/python3.6/site-packages/static_frame/core/type_blocks.py", line 1426, in _extract_iloc
    return self._extract(row_key=key)
  File "/usr/local/lib/python3.6/site-packages/static_frame/core/type_blocks.py", line 1413, in _extract
    shape_reference=self._shape
  File "/usr/local/lib/python3.6/site-packages/static_frame/core/type_blocks.py", line 120, in from_blocks
    raise RuntimeError(f'cannot include array with {block.ndim} dimensions')
RuntimeError: cannot include array with 3 dimensions
```

In debugging I noticed that `IndexGO.loc_to_iloc` is returning an np.int64, which is inconsistent with its type annotation `GetItemKeyType`. Then when `TypeBlocks._extract_array` checks the key type it only checks against `int`, not `INT_TYPES`.

**Comment by flexatone** [comment]
Thank you for isolating this issue! I will take a look.

**Comment by flexatone** [comment]
@abefetterman: Thanks again for reporting this issue and providing your analysis, which proved correct: `INT_TYPES` needed to include `np.integer` not `np.int_` (not for this particular case but the more general case), and `TypeBlocks` needed to use those types for all integer identification. 

This fix is in and will be included in the 0.3.7 release, which I expect to release no later than tomorrow morning. 

Please feel free to post any other issues you find: we greatly appreciate the feedback!
